data/src/test/java/com/linkedin/data/schema/compatibility/TestAnnotationCompatibilityChecker.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.data.schema.compatibility;
+
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.PathSpec;
+import com.linkedin.data.schema.annotation.PegasusSchemaAnnotationHandlerImpl;
+import com.linkedin.data.schema.annotation.SchemaAnnotationHandler;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class TestAnnotationCompatibilityChecker
+{
+  @Test(dataProvider = "annotationCompatibilityCheckTestData")
+  public void testCheckCompatibility(Map<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext,
+      Map<String, Object>>> olderResolvedPropertiesMap, Map<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext,
+      Map<String, Object>>> newerResolvedPropertiesMap, List<SchemaAnnotationHandler> handlers, List<SchemaAnnotationHandler.AnnotationCompatibilityResult> expectedResults)
+  {
+    List<SchemaAnnotationHandler.AnnotationCompatibilityResult> results = AnnotationCompatibilityChecker
+        .getCompatibilityResult(olderResolvedPropertiesMap, newerResolvedPropertiesMap, handlers);
+    Assert.assertEquals(results.size(), expectedResults.size());
+    for (int i = 0; i < results.size(); i++)
+    {
+      Assert.assertEquals(results.get(i).isCheckAnnotation(), expectedResults.get(i).isCheckAnnotation());
+      Assert.assertEquals(results.get(i).getMessages().size(), expectedResults.get(i).getMessages().size());
+      List<CompatibilityMessage> actualCompatibilityMessage = (List<CompatibilityMessage>) results.get(i).getMessages();
+      List<CompatibilityMessage> expectCompatibilityMessage = (List<CompatibilityMessage>) expectedResults.get(i).getMessages();
+      for (int j = 0; j < actualCompatibilityMessage.size(); j++)
+      {
+        Assert.assertEquals(actualCompatibilityMessage.get(j).toString(), expectCompatibilityMessage.get(j).toString());
+      }
+    }
+  }
+
+  @DataProvider
+  private Object[][] annotationCompatibilityCheckTestData()
```

Comment:
Sorry for late review, 
A consideration for this function is to make it readable, and reusable.
Would it be possible if we can use some string formatted json to represent the all the datamap or properties.
And use something like DataMapUtil's function to read those json formated string to dataMap objects?

Reply:
Discussed offline, update the test cases in a cleaner way.